cirq/ops/gate_operation.py

```diff
@@ -186,6 +186,19 @@ def _qasm_(self, args: protocols.QasmArgs) -> Optional[str]:
                               qubits=self.qubits,
                               default=None)
 
+    def __mul__(self, other):
+        from cirq.ops.pauli_string import (SingleQubitPauliStringGateOperation,
```

Comment:
Move imports to the top of the file.

Reply:
I believe this is a "hack" to avoid circular dependencies/imports?

See for instance:
https://github.com/quantumlib/Cirq/blob/28a575d332573d4ddd4d056058caf251a06939bb/cirq/ops/pauli_string.py#L120

But to be honest, I haven't tried moving the imports to the top of the file.

Reply:
I have tried it, and indeed everything fails after I moved the imports to the top of the file